client/my-sites/site-settings/podcasting-details/index.jsx

```diff
@@ -349,7 +350,16 @@ class PodcastingDetails extends Component {
 	};
 
 	onCategoryCleared = () => {
-		this.props.updateFields( { podcasting_category_id: '0' } );
+		const { updateFields, clearDirtyFields, submitForm, siteSlug } = this.props;
+
+		updateFields( { podcasting_category_id: '0' }, () => {
+			submitForm();
+		} );
+
+		// Save changed fields and allow redirect
+		clearDirtyFields();
+
+		page.redirect( '/settings/writing/' + siteSlug + '#site-podcasting-settings' );
```

Comment:
It's not really clear to me _in what order_ these operations will happen.

Does this work instead?

```
updateFields( { podcasting_category_id: '0' }, () => {
    clearDirtyFields();
    submitForm();
    page.redirect( '/settings/writing/' + siteSlug );
} );
```